libPhoneNumber/NBPhoneNumberOfflineGeocoder.m

```diff
@@ -0,0 +1,172 @@
+//
+//  NBPhoneNumberOfflineGeocoder.m
+//  libPhoneNumberiOS
+//
+//  Created by Rastaar Haghi on 6/12/20.
+//  Copyright © 2020 Google LLC. All rights reserved.
+//
+
+#import "NBPhoneNumberOfflineGeocoder.h"
+#import "NBGeocoderMetadataHelper.h"
+#import "NBPhoneNumber.h"
+#import "NBPhoneNumberUtil.h"
+
+@implementation NBPhoneNumberOfflineGeocoder {
+ @private
+  NBPhoneNumberUtil *_phoneNumberUtil;
+  NSCache<NSString *, NBGeocoderMetadataHelper *> *_metadataHelpers;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self != nil) {
+    _phoneNumberUtil = NBPhoneNumberUtil.sharedInstance;
+    // NSLocale provides an array of the user's preferred languages, which can be used
+    // to gather the appropriate language code for NBGeocoderMetadataHelper
+    NSString *languageCode = nil;
+    NSArray<NSString *> *languages = [NSLocale preferredLanguages];
+    for (NSString *language in languages) {
+      if (language != nil) {
```

Comment:
No need to do a nil check here. NSArray cannot contain nil value.

Reply:
fixed.